pytorch_lightning/loops/optimizer/optimizer_loop.py

```diff
@@ -69,14 +69,11 @@ def reset(self) -> None:
             self.optim_progress.optimizer_idx = 0
         self.outputs = [[] for _ in range(len(self.trainer.optimizers))]
 
-    def on_run_start(  # type: ignore[override]
-        self, batch: Any, hiddens: Any, optimizers: List[Optimizer], batch_idx: int
-    ) -> None:
+    def on_run_start(self, batch: Any, optimizers: List[Optimizer], batch_idx: int) -> None:  # type: ignore[override]
         self._batch_idx = batch_idx
         self._optimizers = optimizers
 
-    def advance(self, batch: Any, hiddens: Any, *args, **kwargs) -> None:  # type: ignore[override]
-        self._hiddens = hiddens
+    def advance(self, batch: Any, *args, **kwargs) -> None:  # type: ignore[override]
         result = self._run_optimization(
```

Comment:
@awaelchli `_run_optimization` could have a `hiddens` argument which gets passed all the way down to `_training_step`.

Or `_training_step` could access `self._hiddens` directly. What do you prefer?

Is there a reason why you were setting `self._hiddens` here and accessing it again in `_run_optimization`?

context:
In the future, `advance` will be the one to set `self._hiddens = result.hiddens` after `_run_optimization`

Reply:
if a variable is an instance attribute there is no point in passing it down in its own methods. that would only cause confusion and make the reader question the difference between the local variable "hiddens" and the attribute "self._hiddens". until now it had to be like this because of on_run_end returning the output and hidden state.